packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1785,8 +1785,9 @@ class TimePickerDialog extends StatefulWidget {
     this.confirmText,
     this.helpText,
     this.restorationId,
-    this.initialEntryMode = TimePickerEntryMode.dial,
+    required this.initialEntryMode,
```

Comment:
This is a public API so we can't just change an optional parameter with a default to a required parameter. There are applications that aren't setting this so they would fail to compile if we did this.